### PR TITLE
Misc fixes.

### PR DESCRIPTION
**Pull request by Dirbaio**
- **examples: ensure at least 3 sockets to avoid running out (DHCP, DNS, the user's)**
- **Silence some unused warnings on nightly.**

### CODE REVIEW

File: embassy-net/src/lib.rs [hunk 1]
```diff
@@ -416,7 +416,7 @@ impl<D: Driver> Stack<D> {
     /// ```ignore
     /// let config = embassy_net::Config::dhcpv4(Default::default());
     ///// Init network stack
-    /// static RESOURCES: StaticCell<embassy_net::StackResources<2>> = StaticCell::new();
+    /// static RESOURCES: StaticCell<embassy_net::StackResources<3>> = StaticCell::new();
```
Comment (plaes) [inline embassy-net/src/lib.rs:419]: Maybe add comment why 3?

File: examples/rp/.cargo/config.toml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [target.'cfg(all(target_arch = "arm", target_os = "none"))']
-runner = "probe-rs run --chip RP2040"
+#runner = "probe-rs run --chip RP2040"
+runner = "../../sshprobe.sh"
```
Comment (plaes) [inline examples/rp/.cargo/config.toml:3]: Unintended change?

Comment (Dirbaio) [reply examples/rp/.cargo/config.toml:3]: oops

File: embassy-net/src/lib.rs (hunk 1 above)
Comment (Dirbaio) [reply embassy-net/src/lib.rs:422]: done

Comment (vDorst) [reply embassy-net/src/lib.rs:422]: What happens if you don't have enough resources allocated, from a user perspective?
Does the application just panic's or do the tasks fight for a free socket?

Comment (Dirbaio) [reply embassy-net/src/lib.rs:422]: it panics.

Note this PR doesn't change the behavior, it just fixes many examples that had socket count set to 2 from before we had DNS (so it was enough, one for DHCP and one for the user). Now that we have DNS they need 3.

**Pull request merged**